### ISSUE TITLE
How to set the GPT-2 training parameter

### ISSUE THREAD

**Issue by BigFishMaster** [OpenNMT/OpenNMT-tf#373]
Could you tell me the reason to use ```cosine_annealing``` ?  If I change the decay_type from "cosine_annealing" to "noam_decay_v2", a bug will occur beacause the latter do not use "max_step".
```
    return misc.merge_dict(config, {
        "params": {
            "average_loss_in_time": True,
            "optimizer": "AdamOptimizer",
            "weight_decay": 0.01,
            "learning_rate": 2.5e-4,
            "decay_type": "cosine_annealing",
            "decay_params": {
                "max_step": 1000000,
                "warmup_steps": 2000,
            }
        },
```

**Comment by guillaumekln** [comment]
That's the learning rate schedule that is used in the paper. To use `noam_decay_v2`, you should set parameters that are specific to this schedule, namely:

* `model_dim`
* `warmup_steps`

http://opennmt.net/OpenNMT-tf/package/opennmt.utils.decay.html#opennmt.utils.decay.noam_decay_v2
